challenges/cat-photo-gallery/script.js

```diff
@@ -0,0 +1,36 @@
+window.onload = getRandomCatPicture ()
+
+async function getRandomCatPicture () {
+    await fetch("https://cataas.com/cat?json=true")
+    .then(response => response.json())
+    .then(randomCat => {
+        const image = document.querySelector("img");
+        return image.src = "https://cataas.com" + randomCat.url;
+    }).catch(error => "Error: " + error)
+}
+
+document.onreadystatechange = () => {
+    if (document.readyState === "complete") {
+
+        const checkbox = document.querySelector("#checkbox");
+        const input = document.querySelector("#input");
+        input.disabled = true;
+
+        checkbox.addEventListener("change", function () {
+            !checkbox.checked ? input.disabled = true : input.disabled = false;
+        }) 
+
+        document.querySelector("button").addEventListener("click", async function () {
+            if (!input.value) {
+                await getRandomCatPicture();
+            } else {
+                await fetch(`https://cataas.com/cat/${input.value}?json=true`)
```

Comment:
this else part could be externalized into a standalone named function
named for example as: `getRandomTaggedCatPicture`

i would do it this way:
```js
        if (input.value) {
            await getRandomTaggedCatPicture(input.value);
        } else {
            await getRandomCatPicture();
        }
```